### PR TITLE
Fix resource deletion failure caused by quota calculation error when InPlacePodVerticalScaling is turned on

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

e2e test log is https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121895/pull-kubernetes-e2e-kind-alpha-features/1745322068624281600#1:build-log.txt%3A749

kube-controller-manager log is https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/121895/pull-kubernetes-e2e-kind-alpha-features/1745322068624281600/artifacts/kind-control-plane/pods/kube-system_kube-controller-manager-kind-control-plane_adb1f34692d5d1ca5fb847c6cb9a07eb/kube-controller-manager/0.log.20240111-061905

```log
2024-01-11T06:16:48.651436802Z stderr F I0111 06:16:48.651097       1 pvc_protection_controller.go:335] "Got event on PVC" logger="persistentvolumeclaim-protection-controller" pvc="volume-attributes-class-3934/test-claim-1"
2024-01-11T06:16:49.02831192Z stderr F I0111 06:16:49.028163       1 pvc_protection_controller.go:273] "PVC is unused" logger="persistentvolumeclaim-protection-controller" PVC="volume-attributes-class-3934/test-claim-1"
2024-01-11T06:16:49.032005252Z stderr F E0111 06:16:49.031898       1 pvc_protection_controller.go:206] "Error removing protection finalizer from PVC" err="persistentvolumeclaims \"test-claim-1\" is forbidden: exceeded quota: test-quota, requested: gold. volumeattributesclass.storage.k8s.io/persistentvolumeclaims=1, used: gold.volumeattributesclass.storage.k8s.io/persistentvolumeclaims=1, limited: gold.volumeattributesclass.storage.k8s.io/persistentvolumeclaims=1" logger="persistentvolumeclaim-   protection-controller" PVC="volume-attributes-class-3934/test-claim-1"
```

/cc @vinaykul

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix resource deletion failure caused by quota calculation error when InPlacePodVerticalScaling is turned on
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/1287
```

### CODE REVIEW

**Comment by carlory** [comment]
`Kubernetes e2e suite: [It] [sig-api-machinery] ResourceQuota should create a ResourceQuota and capture the life of a persistent volume claim` is successful

**Comment by carlory** [comment]
unrelated tests fails
/test pull-kubernetes-e2e-kind-alpha-features

**Comment by carlory** [comment]
`pull-kubernetes-e2e-kind-alpha-features` job is successful

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go [hunk 1]
```diff
@@ -518,11 +518,9 @@ func CheckRequest(quotas []corev1.ResourceQuota, a admission.Attributes, evaluat
 			if innerErr != nil {
 				return quotas, innerErr
 			}
-			if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && a.GetResource().GroupResource() == corev1.Resource("pods") {
```
[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go]: > That said, I don't think it's correct to allow negative deltas in a request like this, since admission can complete but persistence in etcd can fail. That would allow a request to decrement quota without actually reducing use.

As another concrete example of this, we don't decrement count quota when deleting an object, we wait for the controller to observe the absence of the object and free the count quota

**Comment by carlory** [comment]
Remove failed test `TestPodResourcesResizeWithResourceQuota`
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/122701/pull-kubernetes-unit/1745489552237137920

File: test/e2e/apimachinery/resource_quota.go [hunk 2]
```diff
@@ -528,6 +530,11 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		err = waitForResourceQuota(ctx, f.ClientSet, f.Namespace.Name, quotaName, usedResources)
 		framework.ExpectNoError(err)
 
+		ginkgo.By("Not allowing a PersistentVolumeClaim to be created that exceeds remaining quota")
```
Comment (liggitt) [inline test/e2e/apimachinery/resource_quota.go:533]: is this test change related to the InPlacePodVerticalScaling change? I don't see how it could be, since it's doing a create operation and the bugfix is around update admission

Comment (liggitt) [reply test/e2e/apimachinery/resource_quota.go:533]: If this is an unrelated test change, split this out into a separate PR

**Comment by carlory** [comment]
/cc @liggitt 

I removed unrelated test. Please review again.

**Comment by pacoxu** [comment]
This seems to be a regression bug that is introduced with the https://github.com/kubernetes/kubernetes/pull/102884/files#diff-1f735790ae870cedaf578264abbcab2a23ec3fb19a47c2c451566cd8638bd38aR521（v1.27）.

**Comment by liggitt** [comment]
/lgtm
/approve

I agree dropping the incorrect unit test is a good idea. I don't know if this changes how the in-place feature expected to work with quota and if that bit needs to be redesigned/retested. Can you make sure this is captured in the items considered while in alpha?

**Comment by pacoxu** [comment]
@carlory for CI, you may watch 
- https://testgrid.k8s.io/sig-node-containerd#cos-cgroupv1-inplace-pod-resize-containerd-e2e-serial

Or run `/test pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2` with latest code.

**Comment by carlory** [comment]
if https://github.com/kubernetes/kubernetes/pull/121895/files#diff-e1a46008c4ddbb9f83858cb22ce56ca0f820b78142a1e8abbbb11aa59127a09bR1233 is merged, we will have an e2e test to cover this part.
